integration/setup/setup.go

```diff
@@ -168,43 +173,100 @@ func WrapTestMain(c *awsclient.Client, g *framework.Guest, h *framework.Host, m
 	os.Exit(v)
 }
 
-func installAWSResource() error {
+func installAWSOperator(h *framework.Host) error {
 	var err error
 
-	var l micrologger.Logger
+	var values string
 	{
-		c := micrologger.Config{}
+		c := chartvalues.AWSOperatorConfig{
+			Guest: chartvalues.AWSOperatorConfigGuest{
+				// TODO remove this setting
+				Update: chartvalues.AWSOperatorConfigGuestUpdate{
+					Enabled: true,
+				},
+			},
+			Provider: chartvalues.AWSOperatorConfigProvider{
+				AWS: chartvalues.AWSOperatorConfigProviderAWS{
+					Encrypter: "kms",
+					Region:    env.AWSRegion(),
+				},
+			},
+			Secret: chartvalues.AWSOperatorConfigSecret{
+				AWSOperator: chartvalues.AWSOperatorConfigSecretAWSOperator{
+					IDRSAPub: env.IDRSAPub(),
+					SecretYaml: chartvalues.AWSOperatorConfigSecretAWSOperatorSecretYaml{
+						Service: chartvalues.AWSOperatorConfigSecretAWSOperatorSecretYamlService{
+							AWS: chartvalues.AWSOperatorConfigSecretAWSOperatorSecretYamlServiceAWS{
+								AccessKey: chartvalues.AWSOperatorConfigSecretAWSOperatorSecretYamlServiceAWSAccessKey{
+									ID:     env.GuestAWSAccessKeyID(),
+									Secret: env.GuestAWSAccessKeySecret(),
+									Token:  env.GuestAWSAccessKeyToken(),
+								},
+								HostAccessKey: chartvalues.AWSOperatorConfigSecretAWSOperatorSecretYamlServiceAWSAccessKey{
+									ID:     env.HostAWSAccessKeyID(),
+									Secret: env.HostAWSAccessKeySecret(),
+									Token:  env.HostAWSAccessKeyToken(),
+								},
+							},
+						},
+					},
+				},
+			},
+			RegistryPullSecret: env.RegistryPullSecret(),
+		}
 
-		l, err = micrologger.New(c)
+		values, err = chartvalues.NewAWSOperator(c)
 		if err != nil {
 			return microerror.Mask(err)
 		}
+
 	}
 
-	o := func() error {
-		// NOTE we ignore errors here because we cannot get really useful error
-		// handling done. This here should anyway only be a quick fix until we use
-		// the helm client lib. Then error handling will be better.
-		framework.HelmCmd("delete --purge aws-config-e2e")
```

Comment:
Not sure either. Each test is in a fresh cluster so the delete shouldn't be needed.

Could just be because the tests here are older. I think the approach in `azure-operator` with the `Resources` function that installs everything is the right one.

Reply:
This is useful for local executions, you need proper teardown in order to rerun.

Reply:
Ah that makes sense. Being able to run tests locally helps a lot when changing them.

The `teardown` logic should cover deleting these resources. When running locally could we run `teardown` before `setup`?

Reply:
You are right, we should rely on the teardown logic and not remove resources on setup. I guess the reason why this was added here is the current teardown approach https://github.com/giantswarm/aws-operator/blob/master/integration/teardown/teardown.go#L30-L60 if something goes wrong removing one of the first resources the rest of them can be left behind because of the early return. We could change this IMO.

Reply:
Totally agree. If we can make `setup` and `teardown` both idempotent that would improve things a lot IMO.

Reply:
Addressed here https://github.com/giantswarm/aws-operator/pull/1112